server/src/main/java/io/crate/role/Role.java

```diff
@@ -165,11 +165,21 @@ public Role with(Set<Privilege> privileges) {
         return new Role(name, new RolePrivileges(privileges), grantedRoles, properties, false);
     }
 
-    public Role with(SecureHash password) {
-        return new Role(name, privileges, grantedRoles, new Properties(properties.login, password, properties.jwtProperties), false);
+    public Role with(@Nullable SecureHash password,
+                     boolean resetPassword,
+                     @Nullable JwtProperties jwtProperties,
+                     boolean resetJwtProperties) {
+        var newPassword = password != null ? password : (resetPassword ? null : properties.password);
+        var newJwtProperties = jwtProperties != null ? jwtProperties : (resetJwtProperties ? null : properties.jwtProperties);
+        return new Role(
```

Comment:
I think the `reset` flag logic should be applied on the call-site, not here because `someRole.with(newPassword, false)` reads like "create a new role with password - but wait - not really, leave it as is"

Reply:
is it okay that on planning we might get a stale cluster state?

Reply:
With call site I mean the transport action, not the planner

Reply:
sorry, what do you mean on planning? cluster state is modified during execution.
And why stale?

Reply:
> what do you mean on planning? 

I thought taking existing values ie role.password(), role.jwtProperties() from ^ should be taken already in planner and `newSecureHash`, `newJwtProperties` should be finalized before sending a request. 

> With call site I mean the transport action/task
https://github.com/crate/crate/pull/15646/commits/0010d5cdd5b1f04f66ea5754ebae1c23d3939fde does that